tracer/src/Datadog.Trace/PlatformHelpers/ContainerMetadata.cs

```diff
@@ -20,7 +20,9 @@ internal static class ContainerMetadata
     {
         private const string ControlGroupsFilePath = "/proc/self/cgroup";
         private const string ContainerRegex = @"[0-9a-f]{64}";
-        private const string UuidRegex = @"[0-9a-f]{8}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{12}";
+        // The second part is the PCF/Garden regexp. We currently assume no suffix ($) to avoid matching pod UIDs
+        // See https://github.com/DataDog/datadog-agent/blob/7.40.x/pkg/util/cgroups/reader.go#L50
+        private const string UuidRegex = @"[0-9a-f]{8}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{12}|([0-9a-f]{8}(-[0-9a-f]{4}){4}$)";
```

Comment:
Is there a reason why this one lacks the non-capture group thing `?:` that is in the other libraries regex?
 
e.g., the other regex compared to ours
- `[0-9a-f]{8}(?:-[0-9a-f]{4}){4}$`
- `[0-9a-f]{8}(-[0-9a-f]{4}){4}$`

Reply:
Huh. Good catch! I didn't notice that before (I copied from [the original](https://github.com/DataDog/datadog-agent/blob/7.40.x/pkg/util/cgroups/reader.go#L51))

We should also use non-capturing. Though I would have said _both_ groups should be non-capturing, seeing as we build the capture later on 🤔 

```
(?:[0-9a-f]{8}(?:-[0-9a-f]{4}){4}$)
```

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/4536/commits/285dc89d09171f64b78017faccd6f4cedeb1a00e